src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -204,6 +204,28 @@ class PythonSetup(Subsystem):
         ),
         advanced=True,
     )
+    no_binary = StrListOption(
+        "--no-binary",
+        help=(
+            "Do not use binary packages (i.e., wheels) for these 3rdparty distributions. Also "
+            "accepts :all: to disable all binary packages. Note that some packages are tricky to "
+            "compile and may fail to install when this option is used on them."
+            "See https://pip.pypa.io/en/stable/cli/pip_install/#install-no-binary for details."
+            'Note: Only takes effect if you use lockfiles, and set lockfile_generator = "pex" '
+            "in the [python] section of your config file."
+        ),
```

Comment:
pedantic but: technically thats not enough. the lockfile must actually be pex. it's possible to have some locks that are requirements.txt style & others pex style. we analyze the lock in pex.py to figure it out

more accurate is

> "the lockfile must be in pex format (set `[pytjon].lockfile_generator = 'pex')

Reply:
nvm! i rereard your impl. this is accurate, but maybeee worth clarifying this option requires regenerating the lock. i could see confusion why changing this wouldn't result in any diff/ behavior until you run henerate-lockfiles